src/tests/test_set_based_verifier.cpp

```diff
@@ -0,0 +1,202 @@
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/od/fastod/model/removal_set.h"
+#include "algorithms/od/set_based_verifier/verifier.h"
+#include "all_csv_configs.h"
+
+namespace tests {
+
+namespace {
+
+using namespace algos::od;
+
+class SetBasedAodVerifierParams {
+private:
+    algos::StdParamsMap params_map_;
+    RemovalSetAsVec expected_removal_set_;
+
+public:
+    SetBasedAodVerifierParams(CSVConfig const& csv_config, config::IndicesType const& oc_context,
+                              config::IndexType oc_left, config::IndexType oc_right,
+                              Ordering ordering, config::IndicesType const& ofd_context,
+                              config::IndexType ofd_right, RemovalSetAsVec removal_set)
+        : params_map_({{config::names::kCsvConfig, csv_config},
+                       {config::names::kOcContext, oc_context},
+                       {config::names::kOcLeftIndex, oc_left},
+                       {config::names::kOcRightIndex, oc_right},
+                       {config::names::kOcLeftOrdering, ordering},
+                       {config::names::kOFDContext, ofd_context},
+                       {config::names::kOFDRightIndex, ofd_right}}),
+          expected_removal_set_(removal_set) {}
+
+    algos::StdParamsMap const& GetParamsMap() const {
+        return params_map_;
+    }
+
+    RemovalSetAsVec const& GetExpectedRemovalSet() const {
+        return expected_removal_set_;
+    }
+};
+
+class TestSetBasedAodVerifier : public ::testing::TestWithParam<SetBasedAodVerifierParams> {};
+
+TEST_P(TestSetBasedAodVerifier, CorrectnessTest) {
+    SetBasedAodVerifierParams const& params(GetParam());
+    algos::StdParamsMap const& params_map = params.GetParamsMap();
+    auto verifier = algos::CreateAndLoadAlgorithm<SetBasedAodVerifier>(params_map);
+    verifier->Execute();
+    EXPECT_EQ(verifier->Holds(), params.GetExpectedRemovalSet().empty());
+
+    RemovalSet actual_removal_set = verifier->GetRemovalSet();
+    RemovalSetAsVec actual_removal_set_vec(actual_removal_set.begin(), actual_removal_set.end());
+    std::sort(actual_removal_set_vec.begin(), actual_removal_set_vec.end());
+    RemovalSetAsVec expected_removal_set_vec = params.GetExpectedRemovalSet();
+    std::sort(expected_removal_set_vec.begin(), expected_removal_set_vec.end());
+    EXPECT_EQ(actual_removal_set_vec, expected_removal_set_vec);
+    SetBasedAodVerifier::Error expected_error =
+            static_cast<SetBasedAodVerifier::Error>(expected_removal_set_vec.size()) /
+            verifier->GetTupleCount();
```

Comment:
since we're specifying type in `static_cast`, we can just use `auto expected_error`